app/controllers/users_controller.rb

```diff
@@ -1,23 +1,23 @@
 class UsersController < ApplicationController
-    before_action :authenticate_user!, only: [:edit, :update]
+  before_action :authenticate_user!, only: [:edit, :update]
 
-    def show
-        @user  = User.find(params[:id])
-        @posts = @user.posts.order(created_at: :desc)
-    end
+  def show
+    @user  = User.find(params[:id])
```

Comment:
shift these queries to model, a single function in user that returns all the posts from the model.
something like. User.get_all_posts(user_id), keep most of the DB interactions in model.